src/main/scala/org/hammerlab/guacamole/assembly/DeBruijnGraph.scala

```diff
@@ -367,4 +381,89 @@ object DeBruijnGraph {
     head ++ rest
   }
 
+  /**
+   * Extract a subsequence from a read sequence
+   *
+   * NOTE: This assumes that the read entirely covers the reference region and does not check for insertions
+   * or deletion
+   *
+   * @param read Read to extract subsequence from
+   * @param startLocus Start (inclusive) locus on the reference
+   * @param endLocus End (exclusive) locus on the reference
+   * @return Subsequence overlapping [startLocus, endLocus)
+   */
+  private def getSequenceFromRead(read: MappedRead, startLocus: Int, endLocus: Int) = {
+    read.sequence.slice(startLocus - read.unclippedStart.toInt, endLocus - read.unclippedStart.toInt)
+  }
+
+  /**
+   * For a given set of reads identify all kmers that appear in the specified reference region
+   *
+   * @param reads  Set of reads to extract sequence from
+   * @param startLocus Start (inclusive) locus on the reference
+   * @param endLocus End (exclusive) locus on the reference
+   * @param minOccurrence Minimum number of times a subsequence needs to appear to be included
+   * @return List of subsequences overlapping [startLocus, endLocus) that appear at least `minOccurrence` time
+   */
+  private def getConsensusKmer(reads: Seq[MappedRead],
```

Comment:
Yes it is. It was used in `discoverHaplotypes` earlier. I'm gone back and forth between using just the reference as source and sinks kmers vs generating a set of from the reads using this vs just using the `sources` and `sinks` from the graphs. Each has their advantages and disadvantages in terms of performance (both variant calling and speed) so still experimenting with which will be best.

Right now using the reference kmers seems fine and faster in germline calling but am testing the alternatives in the somatic caller.